### PR TITLE
Update QueryLogger and write to log all queries with sql statements, parameters and duration of query

### PR DESCRIPTION
**Pull request by mrow4a**
Verified with blackfire:
selection_152

Our Log:
selection_153

Example log:
selection_155

### CODE REVIEW

File: lib/private/DB/Connection.php [hunk 1]
```diff
@@ -165,9 +165,6 @@ public function prepare( $statement, $limit=null, $offset=null ) {
 		$statement = $this->replaceTablePrefix($statement);
 		$statement = $this->adapter->fixupStatement($statement);
 
-		if(\OC::$server->getSystemConfig()->getValue( 'log_query', false)) {
```
Comment (mrow4a) [inline lib/private/DB/Connection.php]: Killed

File: lib/private/legacy/db/statementwrapper.php [hunk 2]
```diff
@@ -64,13 +64,6 @@ public function __call($name,$arguments) {
 	 * @return \OC_DB_StatementWrapper|int
 	 */
 	public function execute($input= []) {
-		if(\OC::$server->getSystemConfig()->getValue( "log_query", false)) {
```
Comment (mrow4a) [inline lib/private/legacy/db/statementwrapper.php]: Start to kill fascinating legacy code?

File: lib/private/Diagnostics/QueryLogger.php [hunk 3]
```diff
@@ -36,6 +36,24 @@ class QueryLogger implements IQueryLogger {
...
+				$sql_string = $query->getSql();
+
+				// Get sql query parameters
+				$params_str = str_replace("\n", " ", var_export($query->getParams(), true));
+
+				// Get duration in msec
+				$int_duration = $query->getDuration() * 1000;
+				$duration = strval($int_duration);
+
+				\OCP\Util::writeLog('core', 'DB CALL : '.$sql_string.' WITH : '.$params_str.' TOOK : '.$duration.' msec',  \OCP\Util::DEBUG);
```
Comment (mrow4a) [inline lib/private/Diagnostics/QueryLogger.php]: Maybe we can write additional APP out of that?

**Comment by mrow4a** [comment]
Any idea how to unit test it and if this is needed?

File: lib/private/Diagnostics/QueryLogger.php [hunk 4]
```diff
@@ -36,6 +36,24 @@ class QueryLogger implements IQueryLogger {
 	 */
 	protected $queries = [];
 
+	public function __destruct() {
+		if(\OC::$server->getSystemConfig()->getValue( "log_query", false)) {
```
Comment (DeepDiver1975) [inline lib/private/Diagnostics/QueryLogger.php]: can we have this is a class of it's own please?

**Comment by mrow4a** [comment]
@DeepDiver1975 @PVince81 Should I adjust some documentation somewhere? This needs few flags

selection_156

**Comment by PVince81** [comment]
@mrow4a config.sample.php

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by DeepDiver1975** [comment]
Thinking one step further: we could do this in an app. And even have special log format/file for the specific needs.

**Comment by mrow4a** [comment]
@DeepDiver1975 So you want to write new default app to the core instead of destructor?

**Comment by DeepDiver1975** [comment]
> Thinking one step further: we could do this in an app. And even have special log format/file for the specific needs.

@PVince81 what do you say?

**Comment by PVince81** [comment]
Wasn't the log_query thing designed to be plugged into by an app ? I don't remember.

**Comment by mrow4a** [comment]
xD

**Comment by DeepDiver1975** [comment]
> Wasn't the log_query thing designed to be plugged into by an app ? I don't remember.

the log_query config switch is really old and only allowed stupid logging to the regular log file.
the querylogger approach was more flexible - this is what clockwork app uses.

we will fully move over to the query logger approach and allow more flexible logging of sql statements.

so - yes - @mrow4a do it as an app :wink:

**Comment by mrow4a** [comment]
Will try to do even more, open API for e.g. jenkins via smashbox to get number of queries for specific requests... ohh it can be wild, but will start simple. @DeepDiver1975 Can I have new table for that or use log file for now?

**Comment by DeepDiver1975** [comment]
> In Table I would like to store number of queries per request and maybe path or sth, so later I can data mine it. In the log I can store everything if someone wants?

I'd start to write this to a file first - possibly an new file with format as needed for the automation - have a chat with @tboerger on whats best.

i'd avoid another table - this add load on the resource which is under monitoring - kind of schroedingers cat ..... :wink:

**Comment by mrow4a** [comment]
As in commit message, make sense or close PR?

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request closed without merging**